src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +166,104 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
+
+        var loggerCache = new Dictionary<string, ILogger>();
+
+        try
+        {
+            // We turned on the JSON formatter for the logger, so we can log the log lines as JSON.
+            await foreach (var batch in resourceLoggerService.WatchAsync(dashboardResourceId).ConfigureAwait(false))
+            {
+                foreach (var logLine in batch)
+                {
+                    try
+                    {
+                        var logMessage = JsonSerializer.Deserialize(logLine.Content, DashboardLogMessageContext.Default.DashboardLogMessage);
+
+                        if (logMessage is not null)
+                        {
+                            var logLevel = logMessage.LogLevel switch
+                            {
+                                "Information" => LogLevel.Information,
+                                "Debug" => LogLevel.Debug,
+                                "Warning" => LogLevel.Warning,
+                                "Error" => LogLevel.Error,
+                                "Critical" => LogLevel.Critical,
+                                "Trace" => LogLevel.Trace,
+                                _ => LogLevel.Information
+                            };
+
+                            if (!loggerCache.TryGetValue(logMessage.Category, out var logger))
+                            {
+                                var categoryTrimmed = logMessage.Category.StartsWith("Aspire.Dashboard.") ?
+                                    logMessage.Category["Aspire.Dashboard.".Length..] : logMessage.Category;
+
+                                loggerCache[logMessage.Category] = logger = loggerFactory.CreateLogger($"Aspire.Hosting.Dashboard.{categoryTrimmed}");
+                            }
+
+                            logger.Log(logLevel, logMessage.EventId, logMessage.Message, null, (s, _) => s);
+                        }
+                    }
+                    catch (JsonException)
+                    {
+                        // Failed to parse, it's not JSON, just log the content as is.
+                        var level = logLine.IsErrorMessage ? LogLevel.Error : LogLevel.Information;
+                        defaultDashboardLogger.Log(level, 0, logLine.Content, null, (s, _) => s);
+                    }
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            defaultDashboardLogger.LogError(ex, "Error logging dashboard logs.");
+        }
+    }
+}
+
+internal class DashboardLogMessage
+{
+    public string Timestamp { get; set; } = string.Empty;
+    public int EventId { get; set; }
+    public string LogLevel { get; set; } = string.Empty;
+    public string Category { get; set; } = string.Empty;
+    public string Message { get; set; } = string.Empty;
+    public JsonObject? State { get; set; }
+}
+
+[JsonSerializable(typeof(DashboardLogMessage))]
+[JsonSourceGenerationOptions(ReadCommentHandling = JsonCommentHandling.Skip)]
```

Comment:
Do we really emit JSON comments in our logs?

Reply:
That's part of this change https://github.com/dotnet/aspire/pull/3523/files#diff-dffe4637b4710f58a20faab4d5a5e45a64e6d6fc71443eb3136286afa68b3c88R154

Reply:
Oh comments, my bad. We don't. I can remove this (copy pasta).

Reply:
Done